cc/base/logging.cc

```diff
@@ -22,7 +22,40 @@
 #include <string.h>
 #include <sys/stat.h>
 #include <sys/types.h>
+#ifdef _WIN32
+// START PATCH
+// HACK: This is not a solution just an attempt at see what is going on and getting PyDP to build on windows.
```

Comment:
This comment looks like it's probably out of date. Can you add a comment that says a little more about what this block does?

Reply:
> Overall, this looks reasonable.

Awesome!

I have added appropriate comments and noticed I'd introduced 2 unused definitions which have been removed.

The function of this code block is to conditionally define code depending on the system Windows vs Unix. This is necessary as these systems have differing system level api's which are included in `<windows.h>` and `<unistd.h>` as an example. 

**Notes:** 
Moving forward it might be easiest to avoid using these system specific methods as recent c++14/17 standards have `#include <filesystem>` which provides a platform independent interface for call to `IS_DIR`, `mkdir` etc... I'm unsure it covers all syscalls required by this library. I have not gone this route as I'm not very familiar with the core of this code base and potential pitfalls. The proposed approach has the nice property that from a unix perspective the code base is unchanged. 

If there is a desire to move to using the more recent c++ standards to avoid platform specific code I'd be happy to research that and it's implications and report back to this project. However, I see that as a different and potentially more invasive change to the on proposed here.